packages/contracts/contracts/extensions/CompliantForwarder/CompliantForwarder.sol

```diff
@@ -0,0 +1,104 @@
+        bytes4 selector = signedFillOrderTransaction.readBytes4(0);
+        if (selector != EXCHANGE_FILL_ORDER_SELECTOR) {
+            revert("EXCHANGE_TRANSACTION_NOT_FILL_ORDER");
+        }
+        
+        // Taker must be compliant
+        require(
+            COMPLIANCE_TOKEN.balanceOf(signerAddress) > 0,
+            "TAKER_UNVERIFIED"
+        );
+
+        // Extract maker address from fill order transaction and ensure maker is compliant
+        // Below is the table of calldata offsets into a fillOrder transaction.
+        /**
+                    ### parameters 
+            0x00      ptr<order>                                                                      
+            0x20      takerAssetFillAmount                                                            
+            0x40      ptr<signature>                                                                  
+                    ### order                                                                           
+            0x60      makerAddress                                                                    
+            0x80      takerAddress                                                                    
+            0xa0      feeRecipientAddress                                                             
+            0xc0      senderAddress                                                                   
+            0xe0      makerAssetAmount                                                                
+            0x100     takerAssetAmount                                                                
+            0x120     makerFee                                                                        
+            0x140     takerFee                                                                        
+            0x160     expirationTimeSeconds                                                           
+            0x180     salt                                                                            
+            0x1a0     ptr<makerAssetData>                                                             
+            0x1c0     ptr<takerAssetData>                                                             
+            0x1e0     makerAssetData                                                                  
+            *         takerAssetData                                                                  
+            *         signature
+            ------------------------------
+            * Context-dependent offsets; unknown at compile time.
+        */
+        // Add 0x4 to a given offset to account for the fillOrder selector prepended to `signedFillOrderTransaction`.
+        // Add 0xc to the makerAddress since abi-encoded addresses are left padded with 12 bytes.
+        // Putting this together: makerAddress = 0x60 + 0x4 + 0xc = 0x70
+        address makerAddress = signedFillOrderTransaction.readAddress(0x70);
```

Comment:
Shouldn't we check the location relative to the order offset? (not an issue if we use `Exchange.fillOrder` ABI)